comptests/s3/test_rest_api_rgw.py

```diff
@@ -256,16 +333,645 @@ def test_user_create_36665(self):
+        self.created_users.append(user_params)
+        self.log.info("END : %s", test_case_name)
+
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-37156')
+    def test_user_create_37156(self):
+        """Test create iam user with same userid but different tenant values."""
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("START: Test create iam user with same userid but different tenant values.")
+        user_name = f"{self.user_name_prefix}1"
+        tenant = "tnt1"
+        tenant2 = "tnt2"
+        user_params = {
+            'display-name': user_name,
+            'uid': user_name,
+            'tenant': tenant
+        }
+        user_params2 = {
+            'display-name': user_name,
+            'uid': user_name,
+            'tenant': tenant2
+        }
+        self.log.info(
+            "Step 1: Creating a new IAM user with name %s", str(user_name))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params))
+        assert status == HTTPStatus.OK, "Not able to create user. Test Failed"
+        self.log.info("Created user details: %s", user_info)
+        self.log.info(
+            "Step 2: Verifying that new IAM user is created successfully")
+        status, user_info = loop.run_until_complete(self.obj.get_user_info(user_params))
+        assert status == HTTPStatus.OK, "Not able to Get user Info. Test Failed"
+        self.log.info("Get user info output: %s", user_info)
+        self.created_users.append(user_params)
+        self.log.info(
+            "Step 3: Creating another IAM user with same name %s but "
+            "different tenant ", str(user_name))
+        loop = asyncio.get_event_loop()
+        status, user_info = loop.run_until_complete(self.obj.create_user(user_params2))
+        assert status == HTTPStatus.OK, "Not able to create user. Test Failed"
+        self.log.info("Created user details: %s", user_info)
+        self.log.info(
+            "Step 4: Verifying another new IAM user is created successfully")
+        status, user_info = loop.run_until_complete(self.obj.get_user_info(user_params2))
+        assert status == HTTPStatus.OK, "Not able to Get user Info. Test Failed"
+        self.log.info("Get user info output: %s", user_info)
+        self.created_users.append(user_params2)
+        self.log.info("END : %s", test_case_name)
+
+    @pytest.mark.api_user_ops
+    @pytest.mark.tags('TEST-37157')
+    def test_user_create_37157(self):
+        """Test create iam user specifying tenant in the uid using $."""
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("START : Test create iam user specifying tenant in the uid using $.")
+        display_name = "test user"
+        user_name = "tenant11$newtestuser"
```

Comment:
are user names with special characters like $ or other allowed ?  If they are allowed does it require any escaping from the client side. We could have those scenarios where we use allowed special chars where they might need special handling from s3 server side.